testsuite/shared/src/main/java/org/wildfly/test/security/common/elytron/ElytronDomainSetup.java

```diff
@@ -0,0 +1,166 @@
+        realmAddress = PathAddress.pathAddress()
+                .append(SUBSYSTEM, ElytronExtension.SUBSYSTEM_NAME)
+                .append("properties-realm", getSecurityRealmName());
+
+        domainAddress = PathAddress.pathAddress()
+                .append(SUBSYSTEM, ElytronExtension.SUBSYSTEM_NAME)
+                .append("security-domain", getSecurityDomainName());
+
+        final ModelNode compositeOp = new ModelNode();
+        compositeOp.get(OP).set(ModelDescriptionConstants.COMPOSITE);
+        compositeOp.get(OP_ADDR).setEmptyList();
+
+        ModelNode steps = compositeOp.get(STEPS);
+
+        // /subsystem=elytron/properties-realm=UsersRoles:add(users-properties={path=users.properties},groups-properties={path=roles.properties})
+        ModelNode addRealm = Util.createAddOperation(realmAddress);
+        addRealm.get("users-properties").get("path").set(getUsersFile());
+        addRealm.get("users-properties").get("plain-text").set(isUsersFilePlain()); // not hashed
+        addRealm.get("groups-properties").get("path").set(getGroupsFile());
+        steps.add(addRealm);
+
+        // /subsystem=elytron/security-domain=EjbDomain:add(default-realm=UsersRoles, realms=[{realm=UsersRoles}])
+        ModelNode addDomain = Util.createAddOperation(domainAddress);
+        addDomain.get("permission-mapper").set("default-permission-mapper"); // LoginPermission for everyone (defined in standalone-elytron.xml)
+        addDomain.get("default-realm").set(getSecurityRealmName());
+        addDomain.get("realms").get(0).get("realm").set(getSecurityRealmName());
+        addDomain.get("realms").get(0).get("role-decoder").set("groups-to-roles"); // use attribute "groups" as roles (defined in standalone-elytron.xml)
+        addDomain.get("realms").get(1).get("realm").set("local");
+        steps.add(addDomain);
+
+        applyUpdate(managementClient.getControllerClient(), compositeOp, false);
+        // TODO: add {"allow-resource-service-restart" => true} to ejbRemoteAddress write-attribute operation once WFLY-8793 / JBEAP-10955 is fixed
+        //       and remove this reload
+        ServerReload.reloadIfRequired(managementClient.getControllerClient());
+    }
+
+    @Override
+    public void tearDown(final ManagementClient managementClient, final String containerId) {
+        // TODO: add {"allow-resource-service-restart" => true} to ejbRemoteAddress write-attribute operation once WFLY-8793 / JBEAP-10955 is fixed
+        //       and remove this reload
+        try {
+            ServerReload.reloadIfRequired(managementClient.getControllerClient());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        // TODO: remove this reload once WFLY-8821 / JBEAP-11074 is fixed
+        try {
+            ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient());
```

Comment:
This should be probably moved after following applyRemoveAllowReload.

Reply:
as WFLY-8821 is related to remoting http-connector, I would say this reload is not needed here at all... will try to remove